### PR TITLE
Deprecate all cx throwable functions

### PR DESCRIPTION
**Pull request by tdejoigny-ledger**
## Description

Deprecate all cx throwable functions

## Changes include

- [X ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Tests
- [X ] Documentation
- [ ] Other (for changes that might not fit in any category)

### CODE REVIEW

File: lib_cxng/include/lcx_aes.h [hunk 1]
```diff
@@ -56,27 +56,12 @@
...
- * @param[out] key     Pointer to the key structure. This must not be NULL.
- *
- * @return             Length of the key.
- *
- * @throw              CX_INVALID_PARAMETER
+ * @deprecated
+ * See #cx_aes_init_key_no_throw
  */
-static inline size_t cx_aes_init_key ( const unsigned char * rawkey, unsigned int key_len, cx_aes_key_t * key )
+__attribute__((deprecated)) static inline size_t cx_aes_init_key ( const unsigned char * rawkey, unsigned int key_len, cx_aes_key_t * key )
```
Comment (xchapron-ledger) [inline lib_cxng/include/lcx_aes.h:64]: could we change this to be a decorator as done here: https://github.com/LedgerHQ/ledger-secure-sdk/commit/27c6ccd2f05c73f405b2e70497cf4a422fd2df8f
This should allow to easily disable / change it depending on compiler or build target or whatever.

File: lib_cxng/include/lcx_aes.h [hunk 2]
```diff
@@ -235,51 +174,9 @@ static inline size_t cx_aes_iv ( const cx_aes_key_t * key, uint32_t mode, unsign
...
- *
- * @return            Length of the output.
- *
- * @throws            CX_INVALID_PARAMETER
- * @throws            INVALID_PARAMETER
- */
-static inline size_t cx_aes ( const cx_aes_key_t * key, uint32_t mode, const unsigned char * in, unsigned int in_len, unsigned char * out, unsigned int out_len )
+ * @deprecated
+ * See #cx_aes_no_throw
+ */static inline size_t cx_aes ( const cx_aes_key_t * key, uint32_t mode, const unsigned char * in, unsigned int in_len, unsigned char * out, unsigned int out_len )
```
Comment (xchapron-ledger) [inline lib_cxng/include/lcx_aes.h:179]: missing new line and attribute?

**Review by xchapron-ledger — COMMENTED** [review]
I'm wondering if we should really drop the comments regarding inputs and outputs?
This will affect the output of https://developers.ledger.com/docs/embedded-app/crypto-api/files/.

Also I'm wondering if we should do the same in `os_seed.h` for `os_perso_derive_xx` functions.

File: lib_cxng/include/lcx_aes.h (hunk 2 above)
Comment (tdejoigny-ledger) [reply lib_cxng/include/lcx_aes.h:179]: OK

File: lib_cxng/include/lcx_aes.h (hunk 1 above)
Comment (tdejoigny-ledger) [reply lib_cxng/include/lcx_aes.h:64]: ok

**Comment by tdejoigny-ledger** [comment]
> I'm wondering if we should really drop the comments regarding inputs and outputs? This will affect the output of https://developers.ledger.com/docs/embedded-app/crypto-api/files/.
> 
> Also I'm wondering if we should do the same in `os_seed.h` for `os_perso_derive_xx` functions.

If the function is deprecated, there is a reference to the new equivalent function to use (where we can find the description of the input/output parameters). We don't want to maintain the input/output of a deprecated function. In addition, other already "deprecated" function are based on this "model".

For `os_perso_derive_xx` functions, yes. Probably better to do that in an other PR

**Review by xchapron-ledger — APPROVED** [review]

**Comment by xchapron-ledger** [comment]
@tdejoigny-ledger Can you please rebase on master so that the Pr can be tested against the OS build and later merged?

**Comment by tdejoigny-ledger** [comment]
> @tdejoigny-ledger Can you please rebase on master so that the Pr can be tested against the OS build and later merged?

@xchapron-ledger done !

**Review by srasoamiaramanana-ledger — APPROVED** [review]

**Pull request merged**